### DISCUSSION TITLE
Not working

### DISCUSSION THREAD

**Discussion by anandbohra** [Dotnet-Boxed/Templates#21] [Q&A]
I am using VS2013 Professional

after downloading latest via NUGET package there are 71 error & 51 warnings
Most of the error is 
Error   35  The type or namespace name 'Boilerplate' could not be found (are you missing a using directive or an assembly reference?)   c:\users\anand.keyline\documents\visual studio 2013\Projects\WebProject2\Services\Sitemap\SitemapService.cs 6   11  WebProject2
Error   38  The type or namespace name 'Autofac' could not be found (are you missing a using directive or an assembly reference?)   c:\users\anand.keyline\documents\visual studio 2013\Projects\WebProject2\App_Start\Startup.Container.cs 6   11  WebProject2
Error   85  The type or namespace name 'SitemapNode' could not be found (are you missing a using directive or an assembly reference?)   c:\users\anand.keyline\documents\visual studio 2013\Projects\WebProject2\Services\Sitemap\SitemapService.cs 85  47  WebProject2

**Comment by RehanSaeed** [comment]
There is a known issue if you create the project more than one folder level away from the root folder where the solution lives. The problem occurs because the HintPath's in the .csproj file which point to the NuGet packages folder become incorrect. It's pretty easy to edit the .csproj and put in one or more extra '..\' to fix the file paths.

Unfortunately, there is not much I can do about that issue, there is a fix coming from SideWaffle (The method used to create the project template) which will fix the HintPath's when the project is created. Not sure when the fix come online but this issue does not affect most users and the fix is simply to create a project one level away from the solution or edit the HintPath's.

Hope That Helps. I'll leave this issue here for others to see.

**Comment by anandbohra** [comment]
I created new project on E: root but again same error

Error   32  The type or namespace name 'Boilerplate' could not be found (are you missing a using directive or an assembly reference?)   E:\WebProject1\App_Start\FilterConfig.cs    5   11  WebProject1

please suggest what to do next since project is almost on E Drive root

**Comment by RehanSaeed** [comment]
I think you have misunderstood me.

# Fix 1
1. File -> New Project
2. Select Web -> ASP.NET MVC Boilerplate
3. Rename the project and change the path but do not change the 'create directory for solution'.
4. Hit ok. This creates the project folder at the right level under the solution folder, where the NuGet packages also live.

# Fix 2

Edit the .csproj file You will see lots of HintPath's like this:

```
   <Reference Include="Boilerplate.Web.Mvc5, Version=1.0.0.0, Culture=neutral, PublicKeyToken=fc5550082a9c642c, processorArchitecture=MSIL">
      <HintPath>..\..\packages\Boilerplate.Web.Mvc5.1.0.18\lib\net45\Boilerplate.Web.Mvc5.dll</HintPath>
      <Private>True</Private>
    </Reference>
```

Notice the `..\..\` in the HintPath. This is the path to your NuGet packages folder from your .csproj file. This path needs to be correct. You need to add as many `..\` as needed to fix the path.

**Comment by anandbohra** [comment]
Tried both the options but none of them worked for me.

Tried on E Drive root
Tried on E Drive root with no solution folder
Added ....\ in both the project 

so in short tried all 4 permutation combination but project failed to build.

please attach working blank solution at your end.

**Comment by RehanSaeed** [comment]
Take a look here for a working sample: 

https://github.com/RehanSaeed/ASP.NET-MVC-Boilerplate/tree/master/Samples

If this is not working for you, something is definitely wrong with your installation. Have you updated the NuGet extension?

**Comment by anandbohra** [comment]
Solution zipped in the the given link worked perfectly in my machine.
Thanks a lot.
